filters/auth/grantconfig.go

```diff
@@ -109,6 +108,8 @@ var (
 	ErrMissingSecretFile      = errors.New("missing secret file")
 	ErrMissingTokeninfoURL    = errors.New("missing tokeninfo URL")
 	ErrMissingProviderURLs    = errors.New("missing provider URLs")
+
+	errMissingSecretsProvider = errors.New("missing secrets provider")
```

Comment:
Why is this not public, which could be used and checked by skipper as library users?

Reply:
I think making those exported was a mistake.
These errors are returned only from `Init()`, how do you think it would be useful for users?

Reply:
Init is called by skipper.run() which is returned by skipper.Run() that is called by library users.

Reply:
Ok, but after this change Init does not always return exported errors but also os.Stat error when file is missing

Reply:
That's true but we can not do anything about it or wrap it, but this does not help.
I think client code could do some:
```
err := skipper.Run(opt)
switch err
case ErrSkipper1:
..
case ErrSkipper2,ErrSkipper3:
..
default:
//handle stdlib error
}
return
```

Of course in most cases they will just log.Fatalf and don't care about the types, but as library we should return typed errors as the best we can.

Reply:
I've made it exported, does not hurt as it won't be use anyway :)